app/helpers/template_helpers.py

```diff
@@ -1,304 +1,378 @@
-        privacy_and_data_protection_path = "preifatrwydd-a-diogelu-data/"
-        terms_and_conditions_path = "telerau-ac-amodau/"
-        contact_us = "cysylltu-a-ni/"
-        languages = "help/ieithoedd-a-hygyrchedd/ieithoedd/"
-        bsl_and_audio_videos = (
-            "help/ieithoedd-a-hygyrchedd/hygyrchedd/fideos-hygyrch-gyda-bsl/"
-        )
-    else:
-        help_path = (
-            "help/help-with-the-questions/online-questions-help/"
-            if is_nisra_theme
-            else "help/how-to-answer-questions/online-questions-help/"
-        )
-        cookies_path = "cookies/"
-        accessibility_statement_path = "accessibility-statement/"
-        privacy_and_data_protection_path = "privacy-and-data-protection/"
-        terms_and_conditions_path = "terms-and-conditions/"
-        contact_us = "contact-us/"
-        languages = "help/languages-and-accessibility/languages/"
-        bsl_and_audio_videos = (
-            "help/languages-and-accessibility/accessibility/accessible-videos-with-bsl/"
-        )
-    urls = {
-        "help": f"{base_url}{help_path}",
-        "cookies": f"{base_url}{cookies_path}",
-        "accessibility_statement": f"{base_url}{accessibility_statement_path}",
-        "privacy_and_data_protection": f"{base_url}{privacy_and_data_protection_path}",
-        "terms_and_conditions": f"{base_url}{terms_and_conditions_path}",
-        "contact_us": f"{base_url}{contact_us}",
+def generate_context(
+    theme: str,
+    language: str,
+    is_post_submission: bool,
+    include_csrf_token: bool,
+) -> dict[str, Any]:
+    return ContextHelper(
+        theme,
+        language,
+        is_post_submission,
+        include_csrf_token,
+        _context_options(theme, language),
+    ).context
+
+
+@lru_cache
+def _context_options(theme: str, language: str) -> ContextOptions:
+    context_options_mapping = {
+        "business": ContextOptions,
+        "health": ContextOptions,
+        "social": ContextOptions,
+        "census": (
+            CensusContextOptionsCymraeg if language == "cy" else CensusContextOptions
+        ),
+        "default": ContextOptions,
+        "census-nisra": CensusNISRAContextOptions,
     }
-    if not is_nisra_theme:
-        urls["languages"] = f"{base_url}{languages}"
-        urls["bsl_and_audio_videos"] = f"{base_url}{bsl_and_audio_videos}"
-
-    return urls
-
-
-def safe_content(content: str) -> str:
-    """Make content safe.
-
-    Replaces variable with ellipsis and strips any HTML tags.
+    return context_options_mapping[theme]()
 
-    :param (str) content: Input string.
-    :returns (str): Modified string.
-    """
-    if content is not None:
-        # Replace piping with ellipsis
-        content = re.sub(r"{.*?}", "…", content)
-        # Strip HTML Tags
-        content = re.sub(r"</?[^>]+>", "", content)
-    return content
 
+def render_template(template: str, **kwargs: Union[str, Mapping]) -> Text:
```

Comment:
How come `Text` is used instead of `str`?

Reply:
It shouldn't be - `str` is correct in this scenario

Reply:
Done